lib/_stream_readable.js

```diff
@@ -1070,13 +1072,84 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+    const src = stream;
+    stream = new Readable({
+      objectMode: true,
+      destroy(err, callback) {
+        destroyImpl.destroyer(src, err);
+        callback();
+      }
+    }).wrap(src);
   }
-  return createReadableStreamAsyncIterator(this);
+
+  const iter = createAsyncIterator(stream);
+  iter.stream = stream;
+  return iter;
 };
 
+async function* createAsyncIterator(stream) {
+  const state = stream._readableState;
+
+  if ((state && state.endEmitted) ||
```

Comment:
Since this is semver-major and won't be backported this can probably be nullish 
```suggestion
  if (state?.endEmitted || state?.closeEmitted) {
```

Reply:
@benjamingr: That fails in a linter test :/

Reply:
```js
/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2842
    throw err
    ^

SyntaxError: Unexpected token (1097:22)
    at Parser.pp$4.raise (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2840:15)
    at Parser.pp.unexpected (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:689:10)
    at Parser.pp$3.parseExprAtom (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2263:12)
    at Parser.pp$3.parseExprSubscripts (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2092:21)
    at Parser.pp$3.parseMaybeUnary (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2069:19)
    at Parser.pp$3.parseExprOps (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2013:21)
    at Parser.pp$3.parseMaybeConditional (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:1996:21)
    at Parser.pp$3.parseMaybeAssign (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:1969:21)
    at Parser.pp$3.parseMaybeConditional (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:2001:30)
    at Parser.pp$3.parseMaybeAssign (/Users/ronagy/GitHub/nxtedition/node/deps/acorn/acorn/dist/acorn.js:1969:21) {
  pos: 34462,
  loc: Position { line: 1097, column: 22 },
  raisedAt: 34463
}
```

Reply:
Bummer, looks like acorn needs an update for that to work which I thought had already happened a while ago :/ Nevermind then